src/AppInstallerCLICore/Workflows/SourceFlow.cpp

```diff
@@ -276,32 +276,30 @@ namespace AppInstaller::CLI::Workflow
     {
         // We are assuming there is only one match, as SourceName is a required parameter.
         const std::vector<Repository::SourceDetails>& sources = context.Get<Data::SourceList>();
-        bool isExplicit = context.Args.Contains(Args::Type::SourceExplicit);
+
         for (const auto& sd : sources)
         {
             // Get the current source with this name.
-            Repository::Source currentSource{ sd.Name };
+            Repository::Source targetSource{ sd.Name };
 
-            // Only permitting Explicit to be edited at this time.
-            Repository::Source source{ sd.Name, isExplicit };
+            // Default to the current explicit value unless we are overriding it.
+            auto isExplicit = sd.Explicit;
+            if (context.Args.Contains(Execution::Args::Type::SourceEditExplicit))
+            {
+                auto explicitArg = context.Args.GetArg(Execution::Args::Type::SourceEditExplicit);
+                isExplicit = Utility::ConvertStringToBool(explicitArg);
+            }
 
-            // Is anything being changed??
-            if (!source.IsEditOfSource(currentSource))
+            Repository::SourceEdit edits{ std::optional<bool>{ isExplicit } };

```

Comment:
Using an `optional` to store the edits means that we should only send a value when the argument is supplied.